File: frontend/src/types/DishInputDTO.ts
import {type AdditionalInformationInputDTO} from "./AdditionalInformationInputDTO.ts";

export type DishInputDTO = {
    name: string;
    price: string;
    type: string;
    imageUrl: string;
    additionalInformation: Record<string, AdditionalInformationInputDTO>;
}

export type DishInputDTOWithId = DishInputDTO & {
    id?: string;
}

File: backend/src/main/java/de/ckollmeier/burgerexpress/backend/dto/DishInputDTO.java
package de.ckollmeier.burgerexpress.backend.dto;

import de.ckollmeier.burgerexpress.backend.interfaces.NamedDTO;
import de.ckollmeier.burgerexpress.backend.interfaces.PricedDTO;

import java.util.Map;

public record DishInputDTO(
        String type,
        String name,
        String price,
        Map<String, AdditionalInformationDTO> additionalInformation,
        String imageUrl
) implements NamedDTO, PricedDTO {
}
